### PR TITLE
python3Packages.array-api-strict: remove with lib for meta

### PR DESCRIPTION
**Pull request by berquist**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/array-api-strict/default.nix [hunk 1]
```diff
@@ -39,11 +39,11 @@ buildPythonPackage rec {
...
 
-  meta = with lib; {
+  meta = {
     homepage = "https://data-apis.org/array-api-strict";
     changelog = "https://github.com/data-apis/array-api-strict/releases/tag/${version}";
     description = "A strict, minimal implementation of the Python array API";
-    license = licenses.bsd3;
-    maintainers = [ maintainers.berquist ];
+    license = lib.licenses.bsd3;
+    maintainers = [ lib.maintainers.berquist ];
```
Comment (SigmaSquadron) [inline pkgs/development/python-modules/array-api-strict/default.nix:47]:
```suggestion
    maintainers = with lib.maintainers; [ berquist ];
```

Optional style change, not a blocker. See the discussion in #331085 and choose whether you want to keep this as-is, or use `with`.

**Review by SigmaSquadron — APPROVED** [review]
diff SGTM

File: pkgs/development/python-modules/array-api-strict/default.nix [hunk 2]
```diff
@@ -37,13 +36,11 @@ buildPythonPackage rec {
     "test_environment_variables"
   ];
 
-  passthru.updateScript = nix-update-script { };
```
Comment (SigmaSquadron) [inline pkgs/development/python-modules/array-api-strict/default.nix:40]: Why?

**Review by SigmaSquadron — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/array-api-strict/default.nix (hunk 2 above)
Comment (berquist) [reply pkgs/development/python-modules/array-api-strict/default.nix:40]: The review in https://github.com/NixOS/nixpkgs/pull/334997#discussion_r1726738619 said that it isn't necessary.

**Review by SigmaSquadron — APPROVED** [review]

File: pkgs/development/python-modules/array-api-strict/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/array-api-strict/default.nix:40]: Yep, that's correct. There is already a default update script.

**Pull request merged**